01_classify_stratified/run_cancer_type_classification.py

```diff
@@ -58,38 +80,55 @@ def process_args():
     else:
         not_in_tcga = set(args.cancer_types) - set(tcga_cancer_types)
         if len(not_in_tcga) > 0:
-            p.error('some cancer types not present in TCGA: {}'.format(
-                ' '.join(not_in_tcga)))
+            parser.error('some cancer types not present in TCGA: {}'.format(
+                         ' '.join(not_in_tcga)))
 
-    return args, sample_info_df
+    # split args into defined argument groups, since we'll use them differently
+    arg_groups = du.split_argument_groups(args, parser)
+    io_args, model_options = arg_groups['io'], arg_groups['model_options']
+
+    # add some additional hyperparameters/ranges from config file to model options
+    model_options.alphas = cfg.alphas
+    model_options.l1_ratios = cfg.l1_ratios
+    model_options.standardize_data_types = cfg.standardize_data_types
+
+    return io_args, model_options, sample_info_df
 
 
 if __name__ == '__main__':
 
     # process command line arguments
-    args, sample_info_df = process_args()
+    io_args, model_options, sample_info_df = process_args()
 
     # create results dir if it doesn't exist
-    args.results_dir.mkdir(parents=True, exist_ok=True)
+    io_args.results_dir.mkdir(parents=True, exist_ok=True)
+    # create subdir for experiment if it doesn't exist
+    experiment_dir = Path(io_args.results_dir, 'cancer_type').resolve()
+    experiment_dir.mkdir(parents=True, exist_ok=True)
+
+    # save model options for this experiment
+    # (hyperparameters, preprocessing info, etc)
+    fu.save_model_options(experiment_dir, model_options)
 
-    # create empty log file if it doesn't exist
+    # create empty error log file if it doesn't exist
     log_columns = [
         'cancer_type',
         'training_data',
         'shuffle_labels',
         'skip_reason'
     ]
-    if args.log_file.exists() and args.log_file.is_file():
-        log_df = pd.read_csv(args.log_file, sep='\t')
+    if io_args.log_file.exists() and io_args.log_file.is_file():
+        log_df = pd.read_csv(io_args.log_file, sep='\t')
     else:
         log_df = pd.DataFrame(columns=log_columns)
-        log_df.to_csv(args.log_file, sep='\t')
+        log_df.to_csv(io_args.log_file, sep='\t')
```

Comment:
Yeah! I usually write warnings/errors to a separate file, since the df isn't really flexible to handle any arbitrary type of warning or error, but the ones I expect to see I put into a dataframe.